.github/workflows/tests-ci.yml

```diff
@@ -70,65 +66,96 @@ jobs:
       - name: Run linter for all packages
         run: npm run lint --ws
 
-      - name: Start dwn-server container
-        run: cd packages/dev-env && docker-compose up -d
-
-      - name: Wait for dwn-server to be ready
-        run: until curl -sf http://localhost:3000/health; do echo -n .; sleep .1; done
+      - name: Run dwn-server (background)
+        run: |
+          node node_modules/@web5/dwn-server/dist/esm/src/main.js &
+          echo "DWN_SERVER_BACKGROUND_PROCESS=$!" >> $GITHUB_ENV
 
       - name: Run tests for all packages
         run: npm run test:node --ws -- --color
         env:
           TEST_DWN_URL: http://localhost:3000
 
+      - name: Terminate dwn-server
+        run: kill $DWN_SERVER_BACKGROUND_PROCESS || true
+
       - name: Upload test coverage to Codecov
         uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d # v3.1.4
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
   test-with-browsers:
+    name: test-with-browsers (group ${{ matrix.group }})
     # Run browser tests using macOS so that WebKit tests don't fail under a Linux environment
     runs-on: macos-latest
+    strategy:
+      # parallelism strategy: agent takes as long as roughly all other pkgs combined.
+      matrix:
+        include:
+          - group: "A"
+            packages: "--workspace packages/agent --workspace packages/dids "
+          - group: "B"
+            packages: "--workspace packages/common --workspace packages/crypto --workspace packages/credentials --workspace packages/user-agent --workspace packages/proxy-agent --workspace packages/api --workspace packages/identity-agent"
     steps:
       - name: Checkout source
-        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        uses: actions/checkout@v4
```

Comment:
[Security scanning](https://github.com/TBD54566975/web5-js/security/code-scanning) checks for pinned dependencies using hash so pin to a v4 using the commit hash of the GitHub action.

> For GitHub workflows used in building and releasing your project, pin dependencies by hash.